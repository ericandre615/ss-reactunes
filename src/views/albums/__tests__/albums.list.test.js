import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AlbumsList from '../list';

const mockState = [{
  collectionId: 328165532,
  artistName: 'Guided By Voices',
  collectionName: 'Suitcase - Failed Experiments and Trashed Aircraft',
  artistViewUrl: 'https://mock.artist.url',
  collectionViewUrl: 'https://mock.collection.url/suitcase',
  artworkUrl100: 'https://mock.album.art/100x100bb.png',
  copyright: 'â„— 2000 Fading Captain Series',
  releaseDate: '2000-12-01T08:00:00Z',
}];

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => mockState), // mockUseSelector,
    useDispatch: jest.fn(),
  };
});

describe('AlbumsList View', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match default snapshot', () => {
    const ShallowAlbumsList = shallow(<AlbumsList />);

    expect(toJson(ShallowAlbumsList)).toMatchSnapshot();
  });
});
