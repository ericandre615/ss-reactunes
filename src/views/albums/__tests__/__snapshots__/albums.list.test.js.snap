// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumsList View should match default snapshot 1`] = `
<div>
  <AlbumList
    albums={
      Array [
        Object {
          "artistName": "Guided By Voices",
          "artistViewUrl": "https://mock.artist.url",
          "artworkUrl100": "https://mock.album.art/100x100bb.png",
          "collectionId": 328165532,
          "collectionName": "Suitcase - Failed Experiments and Trashed Aircraft",
          "collectionViewUrl": "https://mock.collection.url/suitcase",
          "copyright": "â„— 2000 Fading Captain Series",
          "releaseDate": "2000-12-01T08:00:00Z",
        },
      ]
    }
    artworkSize={600}
  />
</div>
`;
